backend/src/services/logbooks-service.js

```diff
@@ -1,232 +1,58 @@
-        operativeNotesXCh,
-        operativeNotesXCm,
-        operativeNotesCAh,
-        operativeNotesCAm,
-        myRole,
-        postOperativeCourse,
-        learningPointsKeyLessons,
-    } = req.body;
-    const { data, error } = await supabase
-        .from("adult_cardiac_logs")
-        .insert({
-            logbook_id: logbookID,
-            type: type,
-            case_no: caseNo,
-            paitent_id: paitentID,
-            surgeon: surgeon,
-            age: age,
-            gender: gender,
-            or_date: orDate,
-            reason: reason,
-            hpi: hpi,
-            social_etoh: socialEtOH,
-            social_drugs: socialDrugs,
-            social_smoking: socialSmoking,
-            social_meds: socialMeds,
-            social_allergies: socialAllergies,
-            pmhx_htn: phmxHTN,
-            pmhx_dmii: phmxDMII,
-            pmhx_dlp: phmxDLP,
-            pmhx_cva: phmxCVA,
-            exam_weight: examWeight,
-            exam_height: examHeight,
-            exam_bmi: examBMI,
-            exam_veins: examVeins,
-            exam_allen_test: examAllenTest,
-            exam_pulses_sixtytwo: examPulsesSixtyTwo,
-            exam_pulses_sixtyfive: examPulsesSixtyFive,
-            exam_pulses_sixtyeight: examPulsesSixtyEight,
-            exam_pulses_seventy: examPulsesSeventy,
-            cath: cathLink,
-            invx_echo_ef: invxEchoEF,
-            invx_echo_rvfx: invxEchoRVFx,
-            invx_wma: invxWMA,
-            invx_aorta: invxAorta,
-            invx_valves: invxValves,
-            invx_cxr: invxCXR,
-            invx_hb: invxHb,
-            invx_w: invxW,
-            invx_pit: invxPit,
-            invx_cr: invxCr,
-            ct: ctLink,
-            surgical_plan: surgicalPlan,
-            surgical_plan_first_operator: surgicalPlanFirstOperator,
-            surgical_plan_issue_or: surgicalPlanIssueOR,
-            surgical_plan_issue_post: surgicalPlanIssuePost,
-            surgical_plan_flag_for_fu: surgicalPlanaFlagForFU,
-            operative_notes_cpb_h: operativeNotesCPBh,
-            operative_notes_cpb_m: operativeNotesCPBm,
-            operative_notes_xc_h: operativeNotesXCh,
-            operative_notes_xc_m: operativeNotesXCm,
-            operative_notes_ca_h: operativeNotesCAh,
-            operative_notes_ca_m: operativeNotesCAm,
-            my_role: myRole,
-            post_operative_course: postOperativeCourse,
-            learning_points_key_lessons: learningPointsKeyLessons,
-        })
-        .select(); 
-    if (error) {
-        throw new Error(error.message);
-    }
-    return data[0];
-}
-
-async function getLogbookType(logbookID, supabase) {
-    const { data, error } = await supabase.from("logbooks").select().eq("id", logbookID);
-    if (error) {
-        throw new Error(error.message);
+        return { error: error.message };
     }
-    return data[0]["type"];
 }
 
 export async function getLogbookLogs(req) {
```

Comment:
We should have try catch statements for getLogbookLogs and getLog as well

Reply:
would they need try catch statements even though getTable returns a error if there's an error? I can see adding an if statement to check the return value of getLogbook type and returning an error then but other than that, would it be necessary to add try catch to both services?